response/max_age.go

```diff
@@ -2,12 +2,47 @@ package response
 
 import (
 	"context"
+	"regexp"
+	"strings"
 
+	"github.com/ONSdigital/dp-legacy-cache-proxy/config"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
 func maxAge(ctx context.Context, path string) int {
 	log.Info(ctx, "Calculating max-age for "+path)
-	// TODO To be implemented as part of DIS-411
-	return 42
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.Fatal(ctx, "error reading the configuration", err)
+		panic(err)
```

Comment:
While working on my next ticket, I noticed that perhaps this wasn't the right thing to do, and instead I should have returned the error and then in `writer.go` I can display a 500 error instead. I have fixed it in this other branch, but I'm happy to backport it here if necessary. Originally, I called `panic` because this should never happen: we've already called `config.Get()` in `main.go`, so if it didn't fail there, it shouldn't fail here. Maybe the right thing to do is to ignore the error?

I'm also open to suggestions on other alternatives. For instance, should the config be read here (like it's right now), or should it be passed as an argument of `maxAge`? Being passed as an argument is probably the right thing to do, as it makes it more testable and less coupled, but then it means that I need to pass it down all the way from `service.go` until here.

Reply:
passing config down the chain doesn't look too onerous (especially given that the BabbageURL is passed similarly, for most of that chain, so could replace/supercede that)

Reply:
ok, can you amend the `panic` to `return 0` perhaps, so it's less explosive, and add a comment to the above effect, thanks

Reply:
I have ended up implementing it here, it didn't take as long as I thought it would 👍

I have also changed the "path" references to "URI", as that's more accurate (we don't just process the path, we also need to take the query parameters into account).